### PR TITLE
Add unit tests for tune.py using pytest

### PR DESCRIPTION
**Pull request by RattataKing**

### CODE REVIEW

File: tuning/test_tune.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import pytest
+from tune import (
+    Configuration, get_contract_tile_sizes, get_pipeline_config, get_shape_dims,
+    get_shapes_mmt, get_shapes_conv, get_shapes_contract, get_shapes_batch_matmul
+)
```
Comment (kuhar) [inline tuning/test_tune.py:5]: Why not import everything `import tune`?

File: tuning/test_tune.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+'''
+Usage: python -m pytest test_tune.py
+'''
+
+def test_get_contract_tile_sizes():
+    config = Configuration(
+        subgroup_size=32,
+        workgroup_size=[16, 16, 1],
+        intrinsic="",
+        tile_sizes=[4, 8, 16],
```
Comment (kuhar) [inline tuning/test_tune.py:16]: Can we try something more realistic with more than 3 dimensions? You can check the actual contracts here: https://github.com/nod-ai/sdxl-scripts/blob/37ef1bc8cc09d9a9afbf05e8faebcc4b331bf269/specs/attention_and_matmul_spec.mlir#L913

File: tuning/test_tune.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+from tune import (
+    Configuration, get_contract_tile_sizes, get_pipeline_config, get_shape_dims,
+    get_shapes_mmt, get_shapes_conv, get_shapes_contract, get_shapes_batch_matmul
+)
+
+'''
+Usage: python -m pytest test_tune.py
+'''
+
+def test_get_contract_tile_sizes():
```
Comment (kuhar) [inline tuning/test_tune.py:11]: Could you also test these tile size functions for mmt and conv?

File: tuning/test_tune.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    ]
+    assert get_shapes_contract(template) == ([2048, 1280], [1280, 1280], [2048, 1280])
+
+def test_get_shapes_batch_matmul():
+    template = [
+        '%10 = linalg.fill ins(%cst : f32) outs(%7 : tensor<1x32x32xf32>) -> tensor<1x32x32xf32>',
+        '%11 = linalg.batch_matmul ins(%8, %9 : tensor<1x32x1024xf32>, tensor<1x1024x32xf32>) outs(%10 : tensor<1x32x32xf32>) -> tensor<1x32x32xf32>',
+        'flow.dispatch.tensor.store %11, %2, offsets = [%arg0, %arg1, %arg2], sizes = [1, 32, 32], strides = [1, 1, 1] : tensor<1x32x32xf32> -> !flow.dispatch.tensor<writeonly:tensor<4x32x64xf32>>'
+    ]
+    assert get_shapes_batch_matmul(template) == ([1, 32, 1024], [1, 1024, 32], [1, 32, 32])
```
Comment (kuhar) [inline tuning/test_tune.py:87]: nit: missing newline

File: tuning/test_tune.py (hunk 2 above)
Comment (RattataKing) [reply tuning/test_tune.py:16]: m,n,k values are already set before Configuration passed to this function, we should cover this test by adding other function

File: tuning/test_tune.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    pass
+
+def test_apply_params_conv():
+    pass
+
+def test_apply_params_contract():
+    pass
+
+def test_apply_params_batch_matmul():
+    pass
```
Comment (kuhar) [inline tuning/test_tune.py:102]: nit: missing newline

**Review by kuhar — APPROVED** [review]
Great, thanks

**Pull request merged**